src/generic/datavgen.cpp

```diff
@@ -1724,6 +1730,7 @@ wxDataViewMainWindow::wxDataViewMainWindow( wxDataViewCtrl *parent, wxWindowID i
     m_useCellFocus = false;
     m_currentRow = (unsigned)-1;
     m_lineHeight = GetDefaultRowHeight();
+    m_heightCache = new wxHashTable();
```

Comment:
Another idea to limit the cache size would be to cache only the position of each `N`-th item, with `N=100` for example, and make `GetLineAt()` smarter by only computing the offset from the last cached item before the given one. But even so it's conceivable for the cache to grow too big, so it looks like we still need some mechanism for pruning it.